emr_run_spark.py

```diff
@@ -108,42 +115,18 @@ def _create_steps(job_flow_name=None,
         'Args': ['unzip', zip_file_on_host, '-d', sources_on_host]
       }
     })
-    if use_mysql:
-      steps.append({
-        'Name': 'setup - jars 1',
-        'ActionOnFailure': 'CANCEL_AND_WAIT',
-        'HadoopJarStep': {
-          'Jar': 'command-runner.jar',
-          'Args': 'touch /home/hadoop/noop.py'.split()
-        }
-      })
-      steps.append({
-        'Name': 'setup - jars 2',
-        'ActionOnFailure': 'CANCEL_AND_WAIT',
-        'HadoopJarStep': {
-          'Jar': 'command-runner.jar',
-          'Args': 'spark-submit --packages mysql:mysql-connector-java:5.1.38 /home/hadoop/noop.py'.split()
-        }
-      })
-      steps.append({
-        'Name': 'setup - jars 3',
-        'ActionOnFailure': 'CANCEL_AND_WAIT',
-        'HadoopJarStep': {
-          'Jar': 'command-runner.jar',
-          'Args': 'sudo cp /home/hadoop/.ivy2/jars/mysql_mysql-connector-java-5.1.38.jar /usr/lib/hadoop/'.split()
-        }
-      })
-    steps.append({
-      'Name': 'run spark {}'.format(spark_main),
-      'ActionOnFailure': 'CANCEL_AND_WAIT',
-      'HadoopJarStep': {
-        'Jar': 'command-runner.jar',
-        'Args': ['spark-submit', '--packages',
-                 'mysql:mysql-connector-java:5.1.38', '--py-files',
-                 zip_file_on_host,
-                 spark_main_on_host] + spark_main_args
-      }
-    })
+    for i in range(num_of_steps):
+        steps.append({
+          'Name': 'run spark {}'.format(spark_main),
+          'ActionOnFailure': 'CANCEL_AND_WAIT',
+          'HadoopJarStep': {
+            'Jar': 'command-runner.jar',
+            'Args': (['spark-submit'] +
+                     packages +
+                     ['--py-files', zip_file_on_host, spark_main_on_host] +
+                     spark_main_args.format(i).split())
```

Comment:
So are you just running the same step `num_of_steps` times?
What am I missing?...